### ISSUE TITLE
Slow Pod Startup Speed with Kata Containers as RuntimeClass in Kubernetes

### ISSUE THREAD

**Issue by wenzhaojie** [kata-containers/kata-containers#7031]
**Before raising this question**

Have you looked at our [document](https://github.com/kata-containers/kata-containers/tree/main/docs) and still haven't found the answer?
Yes！

**Issue **Description**
I have observed significantly slow startup speeds when using Kata Containers as the RuntimeClass in Kubernetes. Even when launching 10 Pods, it takes approximately 20 seconds to complete. This result is not in line with the performance described in the runD paper, which claims the ability to launch 200 sandboxes simultaneously within 1 second, with minimal fluctuations and CPU overhead. Additionally, the paper mentions reduced cold startup latency for a single sandbox at 88ms.

**Reproduction Steps**
Here are the steps I followed to reproduce the issue:

Created a Kubernetes cluster.
Configured Kubernetes to use Kata Containers as the RuntimeClass.
Launched 10 Pods and recorded the startup time.
Observed the startup time for each Pod and the overall startup time.
Expected Results
Based on the information provided in the runD paper, I expected faster Pod startup speeds when using Kata Containers as the RuntimeClass in Kubernetes. Ideally, launching 10 Pods should be completed within 1 second without significant delay fluctuations.

**Actual Results**
However, in my practical testing, I found that using Kata Containers as the RuntimeClass resulted in noticeably slower Pod startup speeds. Launching 10 Pods took approximately 20 seconds, which is inconsistent with the expected results.

**Environment**
Kubernetes Version: 1.27.2
Kata Containers Version: 3.1.2
Operating System: Ubuntu 20.04.6
Other relevant component versions (e.g., containerd): containerd 1.6.12

I'm looking forward to receiving insights and possible solutions to improve the Pod startup speed with Kata Containers in Kubernetes. Thank you!

**Comment by studychao** [comment]
What runtime did you use? Rust runtime or Go runtime? Did you use Dragonball hypervisor?

**Comment by wenzhaojie** [comment]
> What runtime did you use? Rust runtime or Go runtime? Did you use Dragonball hypervisor?

Yes. I use both hypervisor installed with kata-deploy, including clh, fc, dragonball, qemu. Could you please tell me about what is the normal scale up performance of kata-containes? The cold start latency is about？

**Comment by fengwang666** [comment]
@wenzhaojie 10 pods on the same node? What does your pod spec look like? What's the latency to start one pod? I believe the 88ms described in the rund paper requires some special optimizations. From my personal experience, starting a pod with golang kata runtime and clh takes about 2 seconds to start up a dummy pod.
